Assets/Scripts/Wall/WallSegment.cs

```diff
@@ -0,0 +1,89 @@
+using UnityEngine;
+using UnityEngine.Serialization;
+
+namespace Wall
```

Comment:
1) Wall Segment and Scaffolding should depend on an abstraction because they share a lot of similar functions.
2) The WallSegment is not responsible for killing the crossbowmen